bin/mail/2016-09-12-back-to-school/generate_list_functions.rb

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env ruby
+require_relative '../mailing-common/mailing-list-utils'
+require 'active_support'
+require 'active_support/core_ext/object/blank'
+
+def query_dashboard_teachers(query)
+  {}.tap do |results|
+    DASHBOARD_DB.fetch(query).each do |teacher|
+      email = teacher[:email]
+      results[email] = {email: email, name: teacher[:name], id: teacher[:id]}
+    end
+  end
+end
+
+def query_workshop_section_ids
+  DASHBOARD_DB.fetch("SELECT id FROM sections WHERE section_type LIKE '%workshop'").map(:id)
+end
+
+# PD comes in 3 forms: old ops workshops, old K5 workshops, and new pd_workshops.
+# The old ops workshop attendance is tracked in teacher_attendance.
+# For the old K5 and all the new workshops, we can track teachers through section membership.
+def query_all_pd_teachers
+  ops_pd_teachers = query_dashboard_teachers <<-SQL
+    SELECT DISTINCT users.id, users.email, users.name
+    FROM users
+    JOIN workshop_attendance ON (workshop_attendance.teacher_id = users.id)
+    JOIN segments ON (segments.id = workshop_attendance.segment_id)
+    JOIN workshops ON (workshops.id = segments.workshop_id)
+    WHERE users.user_type = 'teacher'
```

Comment:
I suppose we can assume they're teachers if they're in the workshops. I'll verify email instead: `WHERE users.email IS NOT NULL AND users.email <> ''`